src/meta/meta_ckpt.c

```diff
@@ -111,8 +111,18 @@ __wt_meta_checkpoint(
     /* Retrieve the metadata entry for the file. */
     WT_ERR(__wt_metadata_search(session, fname, &config));
 
+/*
+ * Check for the version mismatch in wiredtiger standalone build when compatibility has not been
+ * configured.
+ */
+#ifdef WT_STANDALONE_BUILD
+    if (!F_ISSET(S2C(session), WT_CONN_COMPATIBILITY))
+        /* Check the major/minor version numbers. */
+        WT_ERR(__ckpt_version_chk(session, fname, config));
+#else
     /* Check the major/minor version numbers. */
     WT_ERR(__ckpt_version_chk(session, fname, config));
+#endif
```

Comment:
If we're updating the version numbers in the NOT-standalone build, then this would have to change of course, I think the #else just goes away.